src/main/java/org/shanerx/tradeshop/utils/configuration/SettingSectionKey.java

```diff
@@ -0,0 +1,122 @@
+    TRADE_SHOP_OPTIONS(6, "trade-shop-options", "Trade Shop Options"),
+    ITRADE_SHOP_OPTIONS(7, "itrade-shop-options", "ITrade Shop Options"),
+    BITRADE_SHOP_OPTIONS(8, "bitrade-shop-options", "BiTrade Shop Options"),
+    ILLEGAL_ITEM_OPTIONS(9, "illegal-item-options", "Illegal Item Options", "", "Valid Types: DISABLED, BLACKLIST, WHITELIST\n"),
+    GLOBAL_ILLEGAL_ITEMS(0, ILLEGAL_ITEM_OPTIONS, "global-illegal-items", "", "List for illegal items for both Cost and Product", ""),
+    COST_ILLEGAL_ITEMS(1, ILLEGAL_ITEM_OPTIONS, "cost-illegal-items", "", "List for illegal items for only Cost items", ""),
+    PRODUCT_ILLEGAL_ITEMS(2, ILLEGAL_ITEM_OPTIONS, "product-illegal-items", "", "List for illegal items for only Product items", "");
+
+    private final String key, sectionHeader, valueLead, postComment, preComment;
+    private final int weight; // Lower Weights are printed first
+    private final SettingSectionKey parent;
+
+    SettingSectionKey(int weight, String key, String sectionHeader) {
+        this(weight, null, key, sectionHeader, "", "");
+    }
+
+    SettingSectionKey(int weight, String key, String sectionHeader, String preComment, String postComment) {
+        this(weight, null, key, sectionHeader, preComment, postComment);
+    }
+
+    SettingSectionKey(int weight, SettingSectionKey parent, String key, String sectionHeader) {
+        this(weight, parent, key, sectionHeader, "", "");
+    }
+
+    SettingSectionKey(int weight, SettingSectionKey parent, String key, String sectionHeader, String preComment, String postComment) {
+        this.weight = weight;
+        this.key = key;
+        this.sectionHeader = sectionHeader;
+        this.parent = parent;
+        this.postComment = !postComment.isEmpty() ? postComment : "";
+        this.preComment = !preComment.isEmpty() ? preComment : "";
+        this.valueLead = !key.isEmpty() ? parent != null ? parent.valueLead + "  " : "  " : "";
+    }
+
+    @Override
+    public String getKey() {
+        return !key.isEmpty() ? key : "";
+    }
+
+    @Override
+    public String getPath() {
+        return !key.isEmpty() ? (hasParent() ? parent.getPath() + "." + key : key) : "";
+    }
+
+    @Override
+    public String getFullLead() {
+        return (hasParent() ? parent.getFullLead() : "") + valueLead;
+    }
+
+    @Override
+    public String getFileText() {
+        return parent != null ? parent.valueLead + key : key;
+    }
+
+    @Override
+    public SettingSectionKey getParent() {
+        return parent;
+    }
+
+    @Override
+    public boolean hasParent() {
+        return parent != null;
+    }
+
+    @Override
+    public String getPostComment() {
+        return postComment;
+    }
+
+    @Override
+    public String getPreComment() {
+        return preComment;
+    }
+
+    @Override
+    public String getSectionHeader() {
+        return sectionHeader;
+    }
+
+    @Override
+    public int getWeight() {
+        return weight;
+    }
```

Comment:
This stuff needn't be specific to the settings stuff with a potential similar section for the messages stuff (abstract away to superclass like the things pointed out in the comments below).

Reply:
Enums cant extend other classes which is why this needed such a complex method to do something similar. If I wanted to have an extension of the enums I think I would have to make a custom implementation of enum. From what I could find online this was the best alternative, which is why there are only get calls in the enums(to minimize repetition of complex methods)

Reply:
Enums would not be extended (see the other comment)